app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt

```diff
@@ -100,18 +103,77 @@ object ThreadController {
         return getThreadsQuery(folder, filter).asFlow()
     }
 
+    fun getSearchThreadsAsync(): Flow<ResultsChange<Thread>> {
+        return defaultRealm.query<Thread>("${Thread::isFromSearch.name} == true").asFlow()
+    }
+
     fun getThread(uid: String, realm: TypedRealm = defaultRealm): Thread? {
         return getThreadQuery(uid, realm).find()
     }
 
     fun getThreadAsync(uid: String): Flow<SingleQueryChange<Thread>> {
         return getThreadQuery(uid, defaultRealm).asFlow()
     }
+
+    /**
+     * Init the search threads that we have recovered from the api
+     * - Format remote threads to make it work with the existing logic
+     * - Keep old messages data if it's already exist in local
+     * - Handle duplicates with the existing logic
+     * @param apiThreads The list of api threads that need to be treated
+     * @return a list of search threads
+     */
+    suspend fun initAndGetSearchFolderThreads(apiThreads: List<Thread>): List<Thread> = withContext(Dispatchers.IO) {
+
+        fun MutableRealm.keepOldMessagesAndAddToSearchFolder(thread: Thread, searchFolder: Folder) {
+            thread.messages.forEach { remoteMessage: Message ->
+                MessageController.getMessage(remoteMessage.uid, this)?.let { localMessage ->
+                    val position = thread.messages.indexOfFirst { it.uid == localMessage.uid }
+                    thread.messages[position] = localMessage.copyFromRealm()
+                } ?: run {
+                    remoteMessage.isFromSearch = true
+                }
+                searchFolder.messages.add(remoteMessage)
+            }
+        }
+
+        fun handleDuplicatesMessages(thread: Thread) {
+            if (thread.messages.count() > 1) {
+                val firstMessage = thread.messages.removeAt(0)
+                thread.duplicates = thread.messages
+                thread.messages = realmListOf(firstMessage)
+            }
+        }
```

Comment:
Might be better as an extension fonction of Thread

```suggestion
        fun Thread.handleDuplicatesMessages() {
            if (messages.count() > 1) {
                val firstMessage = messages.removeAt(0)
                duplicates = messages
                messages = realmListOf(firstMessage)
            }
        }
```